### PR TITLE
Interval tree

### PR DESCRIPTION
**Pull request by sstanculeanu**
Added interval tree implementation.

### CODE REVIEW

File: core/tree/utils.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+		return intervals[i].Low <= intervals[j].Low
+	})
+
+	optimizedIntervals := make([]BlocksExceptionInterval, 0, len(intervals))
+	optimizeIntervalsSlice(intervals, &optimizedIntervals)
+
+	return optimizedIntervals
+}
+
+func optimizeIntervalsSlice(intervals []BlocksExceptionInterval, finalIntervals *[]BlocksExceptionInterval) {
```
Comment (iulianpascalau) [inline core/tree/utils.go:173]: I would move this in a new file called optimizedSort.go

File: core/tree/utils.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
+package tree
```
Comment (iulianpascalau) [inline core/tree/utils.go:1]: instead of the name utils.go I suggest to rename this to print.go (carving out the optimization sort, ofc)

Comment (sstanculeanu) [reply core/tree/utils.go:1]: moved

File: core/tree/utils.go [hunk 3]
```diff
@@ -1,188 +1,5 @@
 package tree
```
Comment (iulianpascalau) [inline core/tree/utils.go:1]: remove utils completely?

Comment (sstanculeanu) [reply core/tree/utils.go:1]: removed the file

**Review by iulianpascalau — DISMISSED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**